app/controllers/api/settings_controller.rb

```diff
@@ -1,16 +1,12 @@
 module Api
   class SettingsController < BaseController
-    def show
-      category = @req.c_id
-      selected_sections =
-        if category
-          raise NotFoundError, "Settings category #{category} not found" unless exposed_settings.include?(category)
-          category
-        else
-          exposed_settings
-        end
+    def index
+      render_resource :settings, ::Settings.to_hash.slice(*Array(exposed_settings).collect(&:to_sym))
+    end
 
-      render_resource :settings, ::Settings.to_hash.slice(*Array(selected_sections).collect(&:to_sym))
+    def show
+      raise NotFoundError, "Settings category #{@req.c_id} not found" unless exposed_settings.include?(@req.c_id)
+      render_resource :settings, ::Settings.to_hash.slice(*Array(@req.c_id).collect(&:to_sym))
```

Comment:
Duplicated logic with line 4 for ::Settings.to_hash....   maybe a common method.